includes/elementor-functions.php

```diff
@@ -0,0 +1,219 @@
+<?php
+/**
+ * Hello Elementor functions.
+ *
+ * @package HelloElementor
+ */
+
+use Elementor\Plugin;
+use Elementor\Core\Experiments\Manager as Experiments_Manager;
+
+/**
+ * Register Site Settings Controls.
+ */
+
+add_action( 'elementor/init', 'hello_elementor_settings_init' );
+
+function hello_elementor_settings_init() {
+	require 'controls/settings-header.php';
+	require 'controls/settings-footer.php';
+}
+
+/**
+ * Helper function to return a setting.
+ *
+ * Saves 2 lines to get kit, then get setting. Also caches the kit and setting.
+ *
+ * @param  string $setting_id
+ * @return string|array same as the Elementor internal function does.
+ */
+function hello_elementor_get_setting( $setting_id ) {
+	global $hello_elementor_settings;
+
+	$return = '';
+
+	if ( ! isset( $hello_elementor_settings['active_kit'] ) ) {
```

Comment:
Here you check if this is set. Where is the `'active_kit'` property actually being set?

Reply:
@udidol it's being set below the check, if it's not set it'll set it, and as it's a global, it's only set once, that's the theory right @StuartOB?

Reply:
What's set below the check is `'kit_settings'`, not `'active_kit'`

Reply:
Yes @marc I think your explanation is correct. If I remember the edge case was when I was testing I used this new helper function before Elementor had loaded the global `$hello_elementor_settings` which would bug out, so if it is incorrectly used outside one of our hook locations, and the `$hello_elementor_settings` global is not yet loaded, then we just load it now, before trying to get the setting next step.

Reply:
@udidol - marc has pinted out the inconsistency here with the check, and then whats being set - it;s not correct - well spotted, will check and fix